### PR TITLE
Migrate to 11ty

### LINKED ISSUE #3: Any thoughts on using a static site generator?
We're using a pretty basic setup which is totally OK to begin. But possibly we should use some [static site generator](https://jamstack.org/generators/)? It would scale well.

I can suggest a few from past experience:

- [**11ty**](https://www.11ty.dev): [Simple](https://www.11ty.dev/#quick-start), [extensible](https://www.11ty.dev/docs/plugins/), [offers several templating choices](https://www.11ty.dev/docs/languages/), [data files](https://www.11ty.dev/docs/data-js/), [RSS support](https://www.11ty.dev/docs/plugins/rss/), [Shortcodes](https://www.11ty.dev/docs/shortcodes/)
  Pretty feature-rich, configurable, and not opinionated. Suitable for almost all kinds of sites.
- [**Zola**](https://www.getzola.org/): [Fast](https://www.rust-lang.org/), [Shortcodes](https://www.getzola.org/documentation/content/shortcodes/), [I18n](https://www.getzola.org/documentation/content/multilingual/), [RSS](https://www.getzola.org/documentation/templates/feeds/), [Opinionated structure](https://www.getzola.org/documentation/getting-started/directory-structure/)
  Opinionated, fewer features, and opinionated. More suitable for a blog, in my opinion.

### PR DESCRIPTION
**Pull request by nayaabkhan**
This is a minimal 11ty migration. Mainly to make it easy to review. But, we can do a lot more on top of it, like refactor layouts, fetch the changelog from GitHub, etc.

**Some important notes:**
- I've used [Liquid](https://shopify.github.io/liquid/) mainly because it looks simple and I see it has [most features](https://www.11ty.dev/docs/languages/liquid/#supported-features). But, of course, we can change it.
- Currently, our URLs end with a `.html`. But it is recommended [not to have extensions in URLs](https://www.w3.org/Provider/Style/URI.html). I've used server redirection using surge's [`ROUTER`](https://surge.sh/help/adding-redirects) file that will keep current URLs working (e.g., from SEO).
- Finally, I have also made URLs for the static assets absolute so that they can work on deeply nested pages too.
- Surge requires a paid account when having a `ROUTER` file, so please test it for me 😊

Closes #3

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ airyx_task:
     SURGE_TOKEN: ENCRYPTED[b8c152a2c22ada1210122d08f0cb811b94d5c51380e13002debdd4f558be2b28e89af956b0980db4854ef0e4de850fcb]
   script:
     - pkg update && pkg install -y npm 
-    - npm install -g surge
-    - make -C src
-    - surge --project dist --domain airyx.org
+    - npm ci
```
Comment (nayaabkhan) [inline .cirrus.yml:16]: > It's meant to be used in automated environments such as test platforms, continuous integration, and deployment -- or any situation where you want to make sure you're doing a clean install of your dependencies.

Use clean install on CI: https://docs.npmjs.com/cli/v7/commands/npm-ci.

File: .cirrus.yml [hunk 2]
```diff
@@ -13,7 +13,7 @@ airyx_task:
     SURGE_TOKEN: ENCRYPTED[b8c152a2c22ada1210122d08f0cb811b94d5c51380e13002debdd4f558be2b28e89af956b0980db4854ef0e4de850fcb]
   script:
     - pkg update && pkg install -y npm 
-    - npm install -g surge
```
Comment (nayaabkhan) [inline .cirrus.yml:16]: Because we already have `surge` in `package.json`, we can use the local package instead of installing it globally.

**Comment by mszoek** [comment]
I don't actually have a paid account at Surge either :)
Does this PR play nicely with the redesign work that @llui85 has done in PR #1?

**Comment by nayaabkhan** [comment]
> I don't actually have a paid account at Surge either :)

Oh shoot. Then we can't use `ROUTE` file because it seems they're a paid feature. Let me think of something else.

> Does this PR play nicely with the redesign work that @llui85 has done in PR #1?

Yes, it is compatible with work done in #1. There will be a few conflicts but I can rebase this work on top of it.

I'm also thinking of trying a few other SSGs so there's no hurry pulling this.

**Comment by nayaabkhan** [comment]
> Let me think of something else.

So @mszoek, I can think of these:
1. **Stay with URLs ending with `.html`:** I can use the [11ty permalinks](https://www.11ty.dev/docs/permalinks/) to generate URLs just like how they are now.
2. **Fix the URLs but stay free as in free 🍺:** We can switch to another provider like [Netlify](https://www.netlify.com/) that supports redirects in their free tier.
3. **Fix the URLs by paying surge.**

If you ask me, I would recommend switching to Netlify or similar.

**Comment by mszoek** [comment]
I'm totally good to switch to Netlify.

**Comment by mszoek** [comment]
Cool cool... I'm going to merge #1 now because it's way nicer than what I had. I will set up Netlify later today for the future evolution on this branch.

**Comment by nayaabkhan** [comment]
Going to re-open this in another PR based on the recent changes on `main`.

**Pull request closed without merging**